doc/cabal-commands.rst

```diff
@@ -1,8 +1,48 @@
 cabal-install Commands
 ======================
 
-We now give an in-depth description of all the commands, first describing the
-arguments and flags that are common to some or all of them.
+``cabal help`` groups commands into global, package, new-style project and
+legacy sections. We talk in detail about all of the new-style project commands,
+``help`` and ``list-bin`` but there are other commands.
```

Comment:
@Mikolaj when I started on this, the docs claimed to "give an in-depth description of **all** the commands" but this wasn't so. I included the help command output as a cheap way to show an abbreviated table of contents for `cabal-commands.rst` that we do discuss.

Doing a quick check, it looks like commands `fetch`, `info`, `list`, `user-config`, `upload`, `check`, `get`, `hscolour` and `report` are not discussed. The `init` command is discussed in the quickstart of `developing-packages.rst` so maybe we should link to it? Same goes for the `outdated` and `gen-bounds` commands that are discussed in `cabal-package.rst`.

I see that hscolour is deprecated. Should we move it alongside the v1 stuff (as a separate issue and PR)?

```
$ cabal help hscolour
Generate HsColour colourised code, in HTML format.

Usage: cabal hscolour [FLAGS]

Requires the hscolour program.

Deprecated in favour of 'cabal haddock --hyperlink-source'.
```

I know this is feature creep for this PR, but I've search through https://cabal.readthedocs.io/en/3.6 for "alias" and couldn't find a hit. Would this also be a good time to add a table explaining the forward and legacy aliases?

Reply:
@philderbeast: thank you very much for checking. I wasn't aware we overpromise and underdeliver so excessively. :D

As a first approximation, perhaps just change "all" to something like "some" re described commands? I like the inclusion of the `help` output --- it both works as a table of contents and the demonstration of the first item in the table. Given your findings, indeed, please remove my optimistic assurances that the commands we skip are not recommended. Links to command descriptions in other sections would be great.

Yes, please open a ticket or PR for hscolour! And I think searching in https://cabal.readthedocs.io/en/latest/ may be a good idea (it's regenerated from our CI all the time). I'd open a new ticket (or is here one?) or PR for the aliases.

Reply:
@Mikolaj I side-stepped trouble I had previewing these changes by copying the `cabal-commands.rst` over to the drafts folder of the hakyll setup I have for my blog where pandoc can render a preview for me. Might we be able to setup a gitpod link or codespaces link so that contributors can work on the docs without having to do a local setup (I'm having some python trouble setting up sphinx even though I've had this going locally in the past.).

I will open other issues or PRs as you suggest.

Reply:
> As a first approximation, perhaps just change "all" to something like "some" re described commands?

I made that change @Mikolaj.

Reply:
I've spun off two separate issues from this thread, #7975 and #7978.

Reply:
Splendid. Please let me know and I will re-review.

Reply:
I think I've included each suggestion (but not resolved conversations - should I?). Thanks @Mikolaj this PR is ready for re-review.

Reply:
Got it, thank you. I guess, we don't have an etiquette re conversations --- personally I wait for the reviewer to resolved as "satisified" instead of resolving myself as "addressed".